compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -167,7 +167,7 @@ pub(super) fn op_to_const<'tcx>(
         }
     };
     match immediate {
-        Left(ref mplace) => to_const_value(mplace),
+        Left(mplace) => to_const_value(&mplace),
```

Comment:
Doesn't this change behavior? Previously we took a reference to the existing `immediate` local, now we move `mplace` to a new location and take a reference to that. Why is that better?

Reply:
I thought that LLVM is smart enough to remove unnecessary moves, but it seems [it is not](https://godbolt.org/z/exWTznxos). I'll revert this change 😓